hydroshare/hydroshare
# anonymous users can view resource info + fix handling for innactive users

### Issues addressed
Resolves #3429 
Resolves #3077

### Pull Request Checklist: 















### Positive Test Case
***
#### A. Test with anonymous viewer, active owner:
1. Create a resource with adequate metadata to be made public
2. Make sure access is set to "private" to start
3. Copy the resource landing page URL and paste it into a private browser (not logged in)
4. You will see a simple page that shows the resource name and the owner (instead of redirecting to login)
<img width="1165" alt="image" src="https://user-images.githubusercontent.com/17934193/161167632-5d9c63f5-4142-48a8-bfc4-4058cbdd8429.png">
5. In the logged-in browser, now make the resource public
6. Refresh the private browser and you will see all of the information on the resource

***
#### B. Test with anonymous viewer, innactive owner:
1. Use the Mezzanine admin panel to deactivate the owner of the resource used above
2. Refresh the private browser tab and you will be able to see the full resource information with a message regarding the inactive owner.
<img width="942" alt="image" src="https://user-images.githubusercontent.com/17934193/161169016-6dd9ae6d-7df6-44ba-86d1-2c7e783ffbab.png">

3. Now re-activate the owner and change the resource back to private again
4. Deactivate the owner
5. Refresh the private browser window and you will see just the basic resource info and a message regarding the inactive owner.
<img width="1144" alt="image" src="https://user-images.githubusercontent.com/17934193/161168780-f8acbd74-5f18-48a3-aedf-e1be0b4ef117.png">

***
#### C. Test with logged-in viewer, inactive owner:
1. Log in as a standard (non-admin) user in a 3rd browser that doesn't share session with the other 2
2. Navigate to the (now private) resource using the URL
3. You will see just the basic info w/ message about inactive owner
<img width="1062" alt="image" src="https://user-images.githubusercontent.com/17934193/161169356-fe0ff307-fe3e-43a8-b41b-329426cadcbb.png">
4. Reactivate the owner, make the resource public, deactivate the owner
5. You will now see all of the resource info, with warning about inactive user, for the resource using the logged-in user

***
#### D. Test with logged-in viewer, active owner:
1. Using the admin window, re-activate the owner
2. With the logged in standard user, you will now be able to see all info on the resource without warning about inactive author
3. Using the re-activated owner, make the resource private
4. With the standard user, you will only be able to see the basic info of the resource
